src/ServiceJobList.c

```diff
@@ -0,0 +1,143 @@
+//ServiceJobList.c
+
+// SPDX-FileCopyrightText: 2024, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2024, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+rcl_service_t g_serviceListJobs;
+
+ServiceListJobs_Messages g_messages_ListJobs;
+
+typedef motoros2_interfaces__srv__ListJobs_Response ListJobsResponse;
+
+
+static const size_t RESP_NAMES_LIST_INITIAL_SIZE = 64;
+
+
+void Ros_ServiceListJobs_Initialize()
+{
+    MOTOROS2_MEM_TRACE_START(svc_list_jobs_init);
+
+    const rosidl_service_type_support_t* type_support =
+        ROSIDL_GET_SRV_TYPE_SUPPORT(motoros2_interfaces, srv, ListJobs);
+
+    rcl_ret_t ret = rclc_service_init_default(
+        &g_serviceListJobs, &g_microRosNodeInfo.node, type_support, SERVICE_NAME_LIST_JOBS);
+    motoRosAssert_withMsg(ret == RCL_RET_OK,
+        SUBCODE_FAIL_INIT_SERVICE_LIST_JOBS, "Failed to init service (%d)", (int)ret);
+
+    rosidl_runtime_c__String__init(&g_messages_ListJobs.response.message);
+    //NOTE: if/when needed, this will be resized
+    rosidl_runtime_c__String__Sequence__init(
+        &g_messages_ListJobs.response.names, RESP_NAMES_LIST_INITIAL_SIZE);
+    g_messages_ListJobs.response.names.size = 0;
+
+    MOTOROS2_MEM_TRACE_REPORT(svc_list_jobs_init);
+}
+
+void Ros_ServiceListJobs_Cleanup()
+{
+    rcl_ret_t ret;
+    MOTOROS2_MEM_TRACE_START(svc_list_jobs_fini);
+
+    Ros_Debug_BroadcastMsg("Cleanup service " SERVICE_NAME_LIST_JOBS);
+    ret = rcl_service_fini(&g_serviceListJobs, &g_microRosNodeInfo.node);
+    if (ret != RCL_RET_OK)
+        Ros_Debug_BroadcastMsg("Failed cleaning up " SERVICE_NAME_LIST_JOBS " service: %d", ret);
+    rosidl_runtime_c__String__fini(&g_messages_ListJobs.response.message);
+    rosidl_runtime_c__String__Sequence__fini(&g_messages_ListJobs.response.names);
+
+    MOTOROS2_MEM_TRACE_REPORT(svc_list_jobs_fini);
+}
+
+void Ros_ServiceListJobs_Trigger(const void* request_msg, void* response_msg)
+{
+    RCL_UNUSED(request_msg);
+    ListJobsResponse* response = (ListJobsResponse*)response_msg;
+
+    Ros_Debug_BroadcastMsg("%s: enter", __func__);
+
+    //make sure we don't accidentally return old results (in case of errors fi)
+    rosidl_runtime_c__String__assign(&response->message, "");
+    response->result_code = 0;
+    response->names.size = 0;
+
+    //retrieve list of jobs (".JBI")
+    //TODO(gavanderhoorn): should this also look for other files (JBR)?
+    Ros_Debug_BroadcastMsg("%s: updating job file list", __func__);
+    LONG status = mpRefreshFileList(MP_EXT_ID_JBI);
```

Comment:
@ted-miller: would it make sense to include `JBR`s as well? `mpRefreshFileList(..)` has a constant for those, so we could do that.

Reply:
TBH, I don't even know what a JBR is. I can't find any relevant documentation and have never heard of anyone using them.

I'm ok with just allowing JBI